src/js/controller/controller.js

```diff
@@ -431,16 +431,34 @@ Object.assign(Controller.prototype, {
         }
 
         function _pauseWhenNotViewable(viewable) {
+            _removeInstreamStateListener();
             if (!viewable) {
-                _this.pause({ reason: 'viewable' });
-                _model.set('playOnViewable', !viewable);
+                if (_getAdState() === STATE_BUFFERING) {
+                    _this._instreamAdapter.once('state', _pauseFromViewable, _this);
```

Comment:
I understand the need to defer the call to `_this.pause` while buffering, but don't we always want the model "playOnViewable" attribute to be set here?

Other areas in the code code depend on "playOnViewable" being set here regardless of player state.

Reply:
It seems like this logic could be better handled in `_pause` where we are already checking the ad state and perform custom reasoning around what to do when ad playback is active and not paused.

Reply:
```
            const adState = _getAdState();
            if (adState && adState !== STATE_PAUSED) {
               _updatePauseReason(meta);
               if (adState === STATE_BUFFERING) {
                    // TODO: defer call to pauseAd
                    ...
               } else {
                    _api.pauseAd(true, meta);
               }
                return;
            }
```